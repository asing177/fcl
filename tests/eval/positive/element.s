// All bool values should be "true"

global map<account, int> balances = ();
global bool aElemOf;
global bool notElemOf;

global set<int> amounts = {1,2,3};
global bool elemOfSet;
global bool notElemOfSet;
global account b = u'H1tbrEKWGpbPjSeG856kz2DjViCwMU3qTw3i1PqCLz65';
transition initial -> terminal;

@initial
elementTest(account a, int amount) {
  // tests for element of map
  balances = mapInsert(a, amount, balances);
  aElemOf = element(a, balances);
  notElemOf = !element(b, balances);

  // tests for element of set
  amounts = setInsert(amount, amounts);
  elemOfSet = element(amount, amounts);
  notElemOfSet = !element(0, amounts);

  terminate();
}
