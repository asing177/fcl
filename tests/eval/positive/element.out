*** Deltas ***
[ ModifyGlobal
    Name { unName = "balances" }
    (VMap
       (fromList
          [ ( VAccount
                (Address "fwBVDsVh8SYQy98CzYpNPcbyTRczVUZ96HszhNRB8Ve")
            , VNum (NumDecimal 500)
            )
          ]))
, ModifyGlobal Name { unName = "aElemOf" } (VBool True)
, ModifyGlobal Name { unName = "notElemOf" } (VBool True)
, ModifyGlobal
    Name { unName = "amounts" }
    (VSet
       (fromList
          [ VNum (NumDecimal 1)
          , VNum (NumDecimal 2)
          , VNum (NumDecimal 3)
          , VNum (NumDecimal 500)
          ]))
, ModifyGlobal Name { unName = "elemOfSet" } (VBool True)
, ModifyGlobal Name { unName = "notElemOfSet" } (VBool True)
, ModifyState WorkflowState { places = fromList [ PlaceEnd ] }
, Terminate
]

*** Globals ***
fromList
  [ ( Key { unKey = "aElemOf" } , VBool True )
  , ( Key { unKey = "amounts" }
    , VSet
        (fromList
           [ VNum (NumDecimal 1)
           , VNum (NumDecimal 2)
           , VNum (NumDecimal 3)
           , VNum (NumDecimal 500)
           ])
    )
  , ( Key { unKey = "balances" }
    , VMap
        (fromList
           [ ( VAccount
                 (Address "fwBVDsVh8SYQy98CzYpNPcbyTRczVUZ96HszhNRB8Ve")
             , VNum (NumDecimal 500)
             )
           ])
    )
  , ( Key { unKey = "elemOfSet" } , VBool True )
  , ( Key { unKey = "notElemOf" } , VBool True )
  , ( Key { unKey = "notElemOfSet" } , VBool True )
  , ( Key { unKey = "testAddr" }
    , VAccount (Address "fwBVDsVh8SYQy98CzYpNPcbyTRczVUZ96HszhNRB8Ve")
    )
  , ( Key { unKey = "testAddr2" }
    , VAccount (Address "Fw68hsis6JPW4ed3ER1VC959TTf9DqberLxrq5uECKxh")
    )
  , ( Key { unKey = "testAddr3" }
    , VAccount (Address "4Stv5gWGt78D3p919HFcKEschSmyGKGrAX2G35xJgdhw")
    )
  ]
